api/python/quilt3/backends/s3.py

```diff
@@ -0,0 +1,20 @@
+from quilt3.data_transfer import list_url, get_bytes
+
+from .base import PackageRegistryV1
+
+
+class S3PackageRegistryV1(PackageRegistryV1):
+    def list_packages(self):
+        prev_pkg = None
+        for path, _ in list_url(self.pointers_global_dir):
+            pkg = path.rpartition('/')[0]
+            # A package can have multiple versions, but we should only return the name once.
```

Comment:
Does this rely on ordering for de-duplication? Is ordering guaranteed?

Reply:
Yes, [it's guaranteed](https://docs.aws.amazon.com/AmazonS3/latest/dev/ListingKeysUsingAPIs.html):

> List results are always returned in UTF-8 binary order.